src/librustc_interface/queries.rs

```diff
@@ -186,6 +193,46 @@ impl<'tcx> Queries<'tcx> {
         })
     }
 
+    /// Returns a modified ast::Crate with injected instrumentation code, if requested by command
+    /// line option.
+    ///
+    /// The `instrument` pass/query depends on (and implies) `unresolved_expansion`.
+    pub fn instrument(
```

Comment:
Can't this just be added to `configure_and_expand_inner`?

Reply:
Yes, that would be much simpler, but I thought the desire was to evolve to more of the `query` model, and I wanted to stay in line with that migration. (I read in the docs somewhere that this was a refactoring-in-progress.)

I'm happy to do this either way, as long as it's in line with the expectations of the code owners. (Not sure who makes that kind of decision.)

Reply:
The `librustc_interface` queries and `rustc_middle::ty::TyCtxt` queries are different. The former are just an interface for custom compiler drivers, while the latter are the core of incremental compilation.

Reply:
ok great. In that case I'll just do as you proposed and remove the changes from queries. Nice.